modules/apps/user/user-test/src/testIntegration/java/com/liferay/user/service/test/UserLocalServiceTest.java

```diff
@@ -206,6 +290,152 @@ public void testAuthenticationWhenUserDoesNotExist() throws Exception {
+
+			user = _userLocalService.updateUser(user);
+
+			_userLocalService.checkLockout(user);
+
+			Assert.fail("Password policy is not being enforced");
+		}
+	}
+
+	@Test(expected = UserLockoutException.PasswordPolicyLockout.class)
+	public void testCheckLockoutPortalUserWithLDAPPasswordPolicy()
+		throws Exception {
+
+		try (SafeCloseable safeCloseable1 =
+				_setLDAPAuthConfigurationWithSafeCloseable(true);
+			SafeCloseable safeCloseable2 =
+				_updateDefaultPasswordPolicyWithSafeCloseable(
+					passwordPolicy -> passwordPolicy.setLockout(true))) {
+
+			User user = UserTestUtil.addUser();
+
+			user.setLockout(true);
+			user.setLockoutDate(user.getModifiedDate());
+
+			user = _userLocalService.updateUser(user);
+
+			_userLocalService.checkLockout(user);
+		}
+	}
+
+	@Test
+	public void testCheckPasswordExpiredLDAPUserWithLDAPPasswordPolicy()
+		throws Exception {
+
+		try (SafeCloseable safeCloseable1 =
+				_setLDAPAuthConfigurationWithSafeCloseable(true);
+			SafeCloseable safeCloseable2 =
+				_updateDefaultPasswordPolicyWithSafeCloseable(
+					passwordPolicy -> passwordPolicy.setChangeRequired(true))) {
+
+			User user = _createUser(true, "Liferay123");
+
+			_userLocalService.checkPasswordExpired(user);
+
+			user = _userLocalService.fetchUser(user.getUserId());
+
+			Assert.assertFalse(
+				"LDAP user is not bypassing password policy check",
+				user.isPasswordReset());
+		}
+	}
+
+	@Test
+	public void testCheckPasswordExpiredLDAPUserWithoutLDAPPasswordPolicy()
+		throws Exception {
+
+		try (SafeCloseable safeCloseable1 =
+				_setLDAPAuthConfigurationWithSafeCloseable(false);
+			SafeCloseable safeCloseable2 =
+				_updateDefaultPasswordPolicyWithSafeCloseable(
+					passwordPolicy -> passwordPolicy.setChangeRequired(true))) {
+
+			User user = _createUser(true, "Liferay123");
+
+			_userLocalService.checkPasswordExpired(user);
+
+			user = _userLocalService.fetchUser(user.getUserId());
+
+			Assert.assertTrue(
+				"LDAP user is not adhering to password policy check",
+				user.isPasswordReset());
+		}
+	}
+
+	@Test
+	public void testCheckPasswordExpiredPortalUser() throws Exception {
+		try (SafeCloseable safeCloseable1 =
+				_setLDAPAuthConfigurationWithSafeCloseable(true)) {
+
+			User user;
+
+			try (SafeCloseable safeCloseable2 =
+					_updateDefaultPasswordPolicyWithSafeCloseable(
+						passwordPolicy -> passwordPolicy.setChangeRequired(
+							false))) {
+
+				user = UserTestUtil.addUser();
+
+				Assert.assertFalse(user.isPasswordReset());
+			}
+
+			try (SafeCloseable safeCloseable2 =
+					_updateDefaultPasswordPolicyWithSafeCloseable(
+						passwordPolicy -> {
+							passwordPolicy.setChangeRequired(true);
+							passwordPolicy.setCheckSyntax(true);
+						})) {
+
+				_userLocalService.checkPasswordExpired(user);
+
+				user = _userLocalService.fetchUser(user.getUserId());
+
+				Assert.assertTrue(
+					"User should have to reset their password on first login",
+					user.isPasswordReset());
+			}
```

Comment:
Isn't this a duplicate of `testCheckPasswordExpiredLDAPUserWithLDAPPasswordPolicy`? Why is `checkSyntax` enabled in one but not in the other?

Reply:
I removed the `checkSyntax` enabled, because you're right, it wasn't needed.  However, these two tests are for different user types (LDAP user with LDAP PP enabled and Portal user with LDAP PP enabled), even if their outcomes are the same.